catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -260,19 +260,27 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
         addDocsToResults(docs, results);
 
         if (userSpellcheckIsOn && solrSpellcheckHasResults(solrResponse)) {
-          query.set("q", findQueryToResend(query, solrResponse));
+          Collation collation = findQueryToResend(query, solrResponse);
+          query.set("q", collation.getCollationQueryString());
```

Comment:
✏️ Should we do a null check here for the case that the `solrResponse.getSpellCheckResponse().getCollatedResults()` has size 0 inside the `findQueryToResend` method which will return null and then we would get an null exception here.

Reply:
It won't actually get to this line if that is true.  The if statement above verifies that there are collated results

Reply:
Nice. Looks good to me then.